### ISSUE TITLE
Why do the val_loss and val metrics change for frozen model?

### ISSUE THREAD

**Issue by JasseurHadded1** [ultralytics/ultralytics#6948]
### Search before asking

- [x] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello,

I trained a detection model, then a pose model. Following this, I froze all layers except for the pose head. I have expected that we mustn't have any change in detection path. However, I've observed slight variations in validation losses and metrics before mosaic close, and these changes became more noticeable after the mosaic close. Can anyone provide an explanation for this?

![results](https://github.com/ultralytics/ultralytics/assets/122788858/598bf59b-bb6d-48f5-b9c4-ff8ee775acf8)

Thanks,

**Comment by JasseurHadded1** [comment]
I know that we can say the changes are not really noticeable but I have expected no changes (same input + same network = same output!), Maybe the input changes while pre-processing, but anyone knows in which parts we have those changes if it's the case?

**Comment by glenn-jocher** [comment]
@JasseurHadded1 hello! Even with layers frozen during training, you might see slight fluctuations in val_loss and validation metrics due to factors like non-determinism in the order of floating-point operations, minor differences in data preprocessing or augmentation (including random aspects of the mosaic augmentation), or slight numerical instabilities.

Additionally, the unfrozen layers undergoing training could interact in subtle ways with the frozen part due to shared statistics in batch normalization layers. These could account for the variations you're observing.

Remember, while models strive to provide deterministic outputs, in practice, there are often small variances that occur, especially when only certain parts of the network are being updated. 🧊✨

**Comment by JasseurHadded1** [comment]
@glenn-jocher Thanks for this answer, I just have some question:

1) Why did the performance decrease more after the mosaic close?

2) You have mentioned "random aspects of the mosaic augmentation"  as a cause of variations in val losses, isn't the mosaic applied only on train set? How can it be a cause of variations in val losses?

3) Your point on batch normalization raises a question for me: How are batch normalization params computed for a pre-trained model (especially with some frozen layers)?

Thanks,

**Comment by glenn-jocher** [comment]
Hello @JasseurHadded1,

1) The increased performance change after turning off the mosaic augmentation might be due to the model relying on specific features learned with that augmentation present. The absence of these features could lead to less robust predictions.

2) Apologies for the confusion. You're correct that mosaic augmentation is applied during training. Its direct effects wouldn't impact val losses. However, if the model overfits to the augmented training patterns, it might perform differently on the validation set.

3) Batch normalization parameters for a pre-trained model are usually fixed (frozen) along with the other parameters of the frozen layers. They are not updated during further training unless those layers are unfrozen.

Hope that helps clarify your queries!
